linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/ansible/shared.yml

```diff
@@ -0,0 +1,27 @@
+# platform = multi_platform_rhel,multi_platform_fedora
+# reboot = false
+# strategy = restrict
+# complexity = high
+# disruption = medium
+- name: "Read list of files with incorrect ownership"
+  shell: "rpm -Va --nofiledigest | grep "^.....\(U\|.G\)" | awk '{print $NF}'"
+  register: files_with_incorrect_ownership
+  failed_when: False
+  changed_when: False
+  check_mode: no
+  tags:
+    @ANSIBLE_TAGS@
+    
+- name: Create list of uniq packages
```

Comment:
I don't think that we need to create a unique list of packages. Should be all.

Reply:
The reason you need it to be uniq is because the list of files potentially
could have multiple files ( hundreds ) that come from the same package, if
you don't uniq it the setperms command will be run multiple times against
the same package. No need to do that.

Reply:
Actually, there is a reason to do it. While they are called the same name e.g. `zlib`, they often install different files when talking about different architecture installs e.g. `i686`, `x86_64`, `noarch`, etc. Also, the `uniq` doesn't work because `rpm -qf` correctly returns the file name. For example:

```
# rpm -qa | grep zlib-1
zlib-1.2.11-8.fc28.x86_64
zlib-1.2.11-8.fc28.i686

# rpm -Va --nodigest | grep libz
.....U...    /usr/lib64/libz.so.1.2.11
.....U...    /usr/lib/libz.so.1.2.11

# rpm -qf /usr/lib/libz.so.1.2.11 /usr/lib64/libz.so.1.2.11
zlib-1.2.11-8.fc28.i686
zlib-1.2.11-8.fc28.x86_64
```